internal/v1/handler.go

```diff
@@ -843,9 +843,22 @@ func (h *Handlers) GetOscapProfiles(ctx echo.Context, distribution Distributions
 }
 
 func (h *Handlers) GetOscapCustomizations(ctx echo.Context, distribution Distributions, profile DistributionProfileItem) error {
-	customizations, err := loadOscapCustomizations(h.server.distributionsDir, distribution, profile)
+	dr := h.server.distroRegistry(ctx)
+
+	distroFile, err := dr.Get(string(distribution))
 	if err != nil {
 		return echo.NewHTTPError(http.StatusBadRequest, err)
 	}
+
+	if distroFile == nil {
+		return echo.NewHTTPError(http.StatusBadRequest, fmt.Errorf("Unable to process distro file"))
+	}
+
+	// TODO: get tailoring customizations with the compliance client
+	customizations, err := processRequest(string(profile), distroFile.OscapDatastream, nil)
```

Comment:
Unfortunately this means a tonne of I/O inside the handler.

I have done a POC for an aws lambda as an alternative, if we want to think about that

Reply:
This is not as fleshed out, but here is the poc for the aws lambda:
https://github.com/osbuild/image-builder/compare/main...kingsleyzissou:image-builder:oscap-lambda?expand=1